.github/workflows/build-matrix.org.yml

```diff
@@ -163,4 +163,17 @@ jobs:
         with:
           name: merged-content-artifact
           path: content.tar.gz
+      - name: "ℹ Save PR metadata"
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/pr.json', JSON.stringify(context.payload.pull_request));
```

Comment:
how will this behave for builds not associated with a PR?

Reply:
It seems this will fail because `prInfoArtifact.id` is not defined (https://github.com/matrix-org/matrix-react-sdk/runs/3805077607?check_suite_focus=true would be an example of that).

We should probably handle this more gracefully.

There is no way to exit a job early, so that means that after the script step we would need to have an `if` for every subsequent step. Not great (but not necessarily an issue, just something to be aware of).

If [`on.workflow_run.branches-ignore`](https://github.com/matrix-org/matrix.org/pull/1088#issuecomment-934611174) does not exist and work, I reckon we have to run the script step in all cases as I don’t see a way to determine whether there was a PR before we try to download the PR metadata artifact.

Reply:
I misread the code this was referring to and thought we were talking about the script step of the `netlify.yml` workflow. Will read the documentation further.